lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+var appmetrics = require('appmetrics');
+var os = require('os');
+var path = require('path');
+
+appmetrics.enable('eventloop');
+var amapi = appmetrics.monitor();
+
+// Global objects to store metric data
+var eventLoopMetrics = {count: 0, averageRunningTotal: 0};
+var cpuMetrics = {system: 0, user: 0, count: 0};
+var httpMetrics = {count: 0};
+var probeList = ["leveldown", "memcached", "mongo", "mysql", "oracle", "oracledb", "postgres", "redis", "socketio", "strong-oracle"];
+var probeMetrics = {};
+var heapMetrics = {};
+var gcMetrics = {used: 0, count: 0};
+var messageMetrics = {};
+
+// Keep track of Cpu Profiling
+var profiling = false;
+var objectTrackingEnabled = false;
+var lastObjectHistogram = {};
+
+// Callback interval
+var callbackInterval = 10000;
+var objectTrackingCallbackInterval = 15000;
+
+// ---------------------------------------------
+// Public functions to be exported
+// ---------------------------------------------
+
+// Metrics API Use function
+function use(callback) {
+    // Set up listeners for all the metrics here
+    initialise();
+
+    // The callback needs to fire every 60 seconds with data from the last 60
+	// seconds
+    setInterval(function() {
+        // Get data from last 60 seconds
+        var eventLoopData = getEventLoopData();
+        var cpuData = getCpuData();
+        var httpData = getHTTPData();
+		var probeData;
+		var heapData = getHeapData();
+		var usedHeap = getGCHeapData();
+		var messageData = getMessageData();
+
+		// Fire callback with data to match the strong-agent docs
+		callback("loop.count", eventLoopData.count);
+		callback("loop.minimum", eventLoopData.minimum);
+		callback("loop.maximum", eventLoopData.maximum);
+		callback("loop.average", eventLoopData.average);
+
+		callback("cpu.system", cpuData.system);
+		callback("cpu.total", cpuData.total);
+		callback("cpu.user", cpuData.user);
+
+		callback("http.connection.count", httpData.count);
+
+		// Callback with probe data of each probe if there have been events
+		for(i in probeList){
+			if (probeMetrics[probeList[i]]){
+				probeData = getProbeData(probeList[i]);
+				callback(probeList[i] + ".count", probeData.count);
+				callback(probeList[i] + ".average", probeData.average);
+				callback(probeList[i] + ".minimum", probeData.minimum);
+				callback(probeList[i] + ".maximum", probeData.maximum);
+			}
+		}
+
+		//if (usedHeap > 0) {
+			callback("gc.heap.used", usedHeap);
+		//}
+
+		callback("heap.total", heapData.total);
+		callback("heap.used", heapData.used);
+
+        if (messageData != null) {
+            callback("messages.out.count", messageData.sent);
+            callback("messages.in.count", messageData.received);  
+        }
+		
+		
+	}, callbackInterval);
```

Comment:
you'll have to `unref()` your timers or node won't exit, you'll probably run into this when you run the tests.